metadata/src/main/java/org/apache/kafka/metadata/storage/ScramParser.java

```diff
@@ -173,7 +173,9 @@ byte[] saltedPassword(byte[] salt, int iterations) throws Exception {
                 return configuredSaltedPassword.get();
             }
             return new ScramFormatter(mechanism).saltedPassword(
-                    configuredPasswordString.get(),
+                    configuredPasswordString
```

Comment:
Similarly, it would be better if we change `configuredPasswordString ` to not using String.